examples/classifier/CL4_apply_DTW_based_kmeans_clustering.py

```diff
@@ -0,0 +1,87 @@
+# Standard library imports
+import sys
+new_path = sys.path[0][:-len("/examples/timeseries")]
+sys.path.append(new_path)
+
+# Third party imports
+import itertools
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+from sklearn import cluster
+
+# Local application imports
+from scifin import timeseries as ts
+from scifin import montecarlo as mc
+from scifin import classifier as cl
+
+# Generate two kinds of processes
+N = 50
+mc_MA = mc.generate_series(n=N, names_base="MA-", series_model=ts.moving_average,
+                         start_date="2000-01-01", end_date="2020-01-01", frequency='M',
+                         cst=1., order=2, coeffs=[0.5, 0.2], sigma=0.4)
+
+mc_Garch = mc.generate_series(n=N, names_base="Garch-", series_model=ts.garch,
+                         start_date="2000-01-01", end_date="2020-01-01", frequency='M',
+                         cst=0.2, order_a=1, coeffs_a=[0.2], order_sig=1, coeffs_sig=[0.1])
+
+# Combining them together
+mc1 = []
+for tsi in mc_MA:
+    mc1.append(tsi)
+for tsi in mc_Garch:
+    mc1.append(tsi)
+
+# Plot
+ts.multi_plot(mc1)
+
+# Shuffling elements
+list_idx = [i for i in range(len(mc1))]
+np.random.shuffle(list_idx)
+mc1 = [mc1[i] for i in list_idx]
+
+# Build Normalized DTW distance matrix based on 'abs'
+obs1 = cl.dtw_distance_matrix_from_ts(mc1, window=10, mode='abs', normalize=True)
+
+# Plot this DTW distance matrix
+sns.heatmap(obs1, cmap='viridis')
+plt.title("Initial DTW distance matrix")
+plt.show()
+
+# Fit the model
+n_clust_range = range(2,4,1)
```

Comment:
1 is not needed here. step is already set to 1 as default

Reply:
Thanks. It's a habit I took. I either write range(4) for [0,1,2,3] or write it with 3 arguments, I find it makes it clearer :).

Reply:
I also find it awkward to see the 1 explicitly written :)

Reply:
You guys don't give up :D. Keep that spirit :).